File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    days = []
    n = len(progresses)
    for i in range(n):
        day = math.ceil((100-progresses[i])/speeds[i])
        days.append(day)
        if days:
            if day <= days[i-1] :
                days[i] = days[i-1]
            else:
                answer.append(days.count(days[i-1]))
        if i == n-1:
            answer.append(days.count(days[i]))        

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        Queue<Integer> q = new ArrayDeque<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int day = (int) Math.ceil((double)(100 - progresses[i]) / speeds[i]);
            q.add(day);
        }
        
        int key = q.poll();
        List<Integer> list = new ArrayList<>();
        int num = 1;
        while(!q.isEmpty()) {
            int day = q.poll();
            if (day <= key) {
                num++;
                
            } else {
                list.add(num);
                key = day;
                num = 1;
            }
        }
        list.add(num);
        
        int[] answer = new int[list.size()];
        for(int i=0; i<list.size(); i++) {
            answer[i] = list.get(i);
        }
    
        return answer;
    }
}